src/bin/facilitator.js

```diff
@@ -88,14 +100,91 @@ program.command('progressRedeem <config> <messageHash>')
       await connected.run(
         configPath,
         async (chainConfig, connection) => {
-          const facilitator = new Facilitator(chainConfig, connection);
+          try {
+            const mosaic = chainConfig.toMosaic(connection);
+            const facilitator = new Facilitator(chainConfig, connection, mosaic);
 
-          await facilitator.progressRedeem(messageHash);
+            await facilitator.progressRedeem(messageHash);
+          } catch (e) {
+            logger.error(`error in progress redeem ${e}`);
+          }
         },
       );
     },
   );
 
+program.command('continuousRedeem <config>'
+  + ' <utilityTokenAddress> <totalRedeemAmount> <redeemer> <beneficiary>'
+  + ' <gasPrice> <gasLimit> <minRedeemAmount> <maxRedeemAmount>')
+  .action(
+    async (
+      configPath,
+      utilityTokenAddress,
+      totalRedeemAmount,
+      redeemer,
+      beneficiary,
+      gasPrice,
+      gasLimit,
+      minRedeemAmount,
+      maxRedeemAmount,
+    ) => {
+      await connected.run(
+        configPath,
+        async (chainConfig, connection) => {
+          try {
+            minRedeemAmount = new BN(minRedeemAmount);
+            maxRedeemAmount = new BN(maxRedeemAmount);
+            totalRedeemAmount = new BN(totalRedeemAmount);
+
+            // To redeem same amount each time, maxRedeem amount is passed zero.
+            if (maxRedeemAmount.isZero()) {
+              maxRedeemAmount = minRedeemAmount;
+            }
+
+            if (maxRedeemAmount.gt(totalRedeemAmount)) {
+              maxRedeemAmount = totalRedeemAmount;
+            }
+
+            if (minRedeemAmount.gt(maxRedeemAmount)) {
+              minRedeemAmount = maxRedeemAmount;
+            }
+            const utilityToken = new ContractInteract.EIP20Token(
+              connection.auxiliaryWeb3,
+              utilityTokenAddress,
+            );
+            const mosaic = chainConfig.toMosaicFromUtilityToken(
+              connection,
+              utilityTokenAddress,
+            );
+            const facilitator = new Facilitator(chainConfig, connection, mosaic);
+            let currentBalance = new BN((await utilityToken.balanceOf(redeemer)));
+
+            if (currentBalance.gt(totalRedeemAmount)) {
+              let amount = randomNumberBetweenRange(minRedeemAmount, maxRedeemAmount);
+              let amountRedeemed = amount;
+              while (amountRedeemed.lten(totalRedeemAmount)) {
```

Comment:
Only issue will be if ```amountRedeemed``` is greater than ```totalRedeemAmount``` in first randomization calculation then redeem will not happen at all 😮 . 
This could be fine since it's a rare case and JLP is a test framework.